pandas/core/internals.py

```diff
@@ -1440,7 +1454,12 @@ def func(cond, values, other):
             if try_cast:
                 result = self._try_cast_result(result)
 
-            return self.make_block(result)
+            if isinstance(result, np.ndarray):
```

Comment:
again this is just completely confusing to do and makes the code way more complex. find a better way